framework/src/modules/interoperability/utils.ts

```diff
@@ -593,3 +618,347 @@ export const commonCCUExecutelogic = async (args: CommonExecutionLogicArgs) => {
+			channelData.storeKey,
+			channelData.storeValue,
+			channelSchema,
+		);
+	}
+
+	const chainValidatorsStoreKeySet = new dataStructures.BufferSet();
+	const chainValidatorsStore = getStore(id, STORE_PREFIX_CHAIN_VALIDATORS);
+	for (const chainValidators of genesisStore.chainValidatorsSubstore) {
+		if (chainValidatorsStoreKeySet.has(chainValidators.storeKey)) {
+			throw new Error(
+				`Chain validators store key ${chainValidators.storeKey.toString('hex')} is duplicated.`,
+			);
+		}
+		chainValidatorsStoreKeySet.add(chainValidators.storeKey);
+
+		const { activeValidators, certificateThreshold } = chainValidators.storeValue;
+		if (activeValidators.length < 1 || activeValidators.length > MAX_NUM_VALIDATORS) {
+			throw new Error(
+				`Active validators must have at least 1 element and at most ${MAX_NUM_VALIDATORS} elements.`,
+			);
+		}
+
+		let totalWeight = BigInt(0);
+		for (let j = 0; j < activeValidators.length; j += 1) {
+			const activeValidator = activeValidators[j];
+
+			const { blsKey } = activeValidator;
+			if (j < activeValidators.length - 1 && blsKey.compare(activeValidators[j + 1].blsKey) >= 0) {
+				throw new Error(
+					'Active validators must be ordered lexicographically by blsKey property and pairwise distinct.',
+				);
+			}
+
+			const { bftWeight } = activeValidator;
+			totalWeight += bftWeight;
+		}
+
+		if (totalWeight > MAX_UINT64) {
+			throw new Error(
+				'The total BFT weight of all active validators has to be less than or equal to MAX_UINT64.',
+			);
+		}
+
+		const checkBftWeightValue = totalWeight / BigInt(3) + BigInt(1);
+		if (checkBftWeightValue > totalWeight || checkBftWeightValue > certificateThreshold) {
+			throw new Error('The total BFT weight of all active validators is not valid.');
+		}
+
+		await chainValidatorsStore.setWithSchema(
+			chainValidators.storeKey,
+			chainValidators.storeValue,
+			chainValidatorsSchema,
+		);
+	}
+
+	const chainDataStoreKeySet = new dataStructures.BufferSet();
+	const chainDataStore = getStore(id, STORE_PREFIX_CHAIN_DATA);
+	let isAnotherSidechainAccount = 0;
+	for (const chainData of genesisStore.chainDataSubstore) {
+		const chainDataStoreKey = chainData.storeKey;
+		if (chainDataStoreKeySet.has(chainDataStoreKey)) {
+			throw new Error(`Chain data store key ${chainDataStoreKey.toString('hex')} is duplicated.`);
+		}
+		chainDataStoreKeySet.add(chainDataStoreKey);
+
+		const chainAccountStatus = chainData.storeValue.status;
+		if (chainAccountStatus === CHAIN_TERMINATED) {
+			if (outboxRootStoreKeySet.has(chainDataStoreKey)) {
+				throw new Error('Outbox root store cannot have entry for a terminated chain account.');
+			}
+			if (
+				!channelDataStoreKeySet.has(chainDataStoreKey) ||
+				!chainValidatorsStoreKeySet.has(chainDataStoreKey)
+			) {
+				throw new Error(
+					`Chain data store key ${chainDataStoreKey.toString(
+						'hex',
+					)} missing in some or all of channel data and chain validators stores.`,
+				);
+			}
+		}
```

Comment:
However, if the chain account has status set to terminated, the entry in the outbox root must not be present.